keps/sig-scheduling/5075-dra-consumable-capacity/README.md

```diff
@@ -0,0 +1,1150 @@
+// Exactly one of the consumption policies must be defined.
+type CapacitySharingPolicy struct {
+   // DiscreteValues defines a set of acceptable quantity values in consuming requests.
+   //
+   // +optional
+   // +oneOf=SharingPolicy
+   DiscreteValues *CapacitySharingPolicyDiscrete
+
+
+   // ValueRange defines an acceptable quantity value range in consuming requests.
+   //
+   // +optional
+   // +oneOf=SharingPolicy
+   ValueRange *CapacitySharingPolicyRange
+}
+
+// CapacitySharingPolicyDiscrete defines a set of discrete allowed capacity values.
+// If Options is not provided, only default value is valid.
+//
+// - If the requested amount is not listed in the options, it is rounded up to the next higher valid value.
+// - If the requested amount exceeds the maximum value in the available options, the request does not satisfy the policy,
+//   and the device cannot be allocated.
+type CapacitySharingPolicyDiscrete struct {
+    // Default specifies the default capacity to be used for a consumption request
+    // if no value is explicitly provided.
+    //
+    // +required
+    Default resource.Quantity
+
+    // Options defines a list of additional valid capacity values that can be requested.
+    // The Default must not be one of the Options.
+    //
+    // +optional
+    // +listType=atomic
+    Options []resource.Quantity
+}
+
+
+// CapacitySharingPolicyRange defines a valid range for consumable capacity values.
+//
+// - If the requested amount is less than Minimum, it is rounded up to the Minimum value.
+// - If the requested amount is between Minimum and Maximum, ChunkSize is set,
+//   and the amount does not align with the ChunkSize,
+//   it will be rounded up to the next value matching Minimum + (n * ChunkSize).
+// - If the requested or rounded amount exceeds Maximum (if set), the request does not satisfy the policy,
+//   and the device cannot be allocated.
+//
+// - If ChunkSize is not set, the requested amount is used as-is, provided it falls within the range.
+type CapacitySharingPolicyRange struct {
+    // Minimum specifies the minimum capacity allowed for a consumption request.
+    // This also acts as the default if no value is specified.
+    //
+    // +required
+    Minimum resource.Quantity
+
+    // Maximum defines the upper limit for capacity that can be requested.
+    //
+    // +optional
+    Maximum *resource.Quantity
+
+    // ChunkSize defines the step size between valid capacity amounts within the range.
+    // If set, requested amounts are rounded up to the nearest multiple of ChunkSize from the Minimum.
+    //
+    // +optional
+    ChunkSize *resource.Quantity
+}
+```
+
+#### ResourceClaimSpec's DeviceRequest
+
+```go
+
+type DeviceRequest struct {
+...
+   // CapacityRequests define resource requirements against each capacity.
+   //
+   // +optional
+   // +featureGate=DRAConsumableCapacity
+   CapacityRequests *CapacityRequirements
```

Comment:
this is going under DeviceRequest, and not inside ExactDeviceRequest / DeviceSubRequest?

I thought that `FirstAvailable []DeviceSubRequest` allowed selection of wildly different devices ... if so, the CapacityRequests could be very different depending on which subrequest was matched, right?

Reply:
@liggitt Thank you. I need to update the KEP according to the change on API introduced by [KEP 4816](https://github.com/kubernetes/enhancements/tree/4c66da9cbf355ff618642aea3d98d17111311646/keps/sig-scheduling/4816-dra-prioritized-list). Here is the commit: https://github.com/kubernetes/enhancements/pull/5104/commits/55fa6bf17edb358bf49ddcd4c22de65938b0082c.

Reply:
To answer @liggitt's question: I think this belongs into `ExactDeviceRequest` and `DeviceSubRequest` as in the KEP now, because then a prioritized list can have different alternatives which differ in how much capacity they request.

Having `CapacityRequests` in `DeviceRequest` would force all of them to use same capacity.